vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -359,7 +359,7 @@ func (feeder *clusterStateFeeder) LoadVPAs() {
 		}
 
 		selector, conditions := feeder.getSelector(vpaCRD)
-		klog.V(4).Infof("Using selector %s for VPA %s/%s", selector.String(), vpaCRD.Namespace, vpaCRD.Name)
```

Comment:
For some lines above you added that they are namespace/name. Is there a specific rule when to print namespacedName vs. the whole resource?

Reply:
We are not printing the whole resource. As explained in https://github.com/kubernetes/autoscaler/pull/6909#discussion_r1639445690 `klog.KObj` only prints `<namespace>/<name>`.

If you are asking why in some places we use `klog.KObj` and in some places we have to use `%s/%s` and pass the namespace and name: `KObj` accepts the interface `KMetadata`: https://github.com/kubernetes/autoscaler/blob/f03e98f44714de186b7ce7fcfb38ed031cf6f673/vertical-pod-autoscaler/vendor/k8s.io/klog/v2/k8s_references.go#L72-L75. Structs like https://github.com/kubernetes/autoscaler/blob/8bc327e86899cae17b3d7e1170fc10d7be1221a0/vertical-pod-autoscaler/pkg/recommender/model/types.go#L160-L163 and https://github.com/kubernetes/autoscaler/blob/8bc327e86899cae17b3d7e1170fc10d7be1221a0/vertical-pod-autoscaler/pkg/recommender/model/types.go#L144-L150 don't implement that interface and that's why I am using `%s/%s` for them.

Reply:
I see, thanks. I wonder if it's worth implementing that interface on those types instead for consistency. But I'm good with this too.

Reply:
> I wonder if it's worth implementing that interface on those types instead for consistency. But I'm good with this too.

I discovered [`klog.KRef`](https://github.com/kubernetes/autoscaler/blob/f03e98f44714de186b7ce7fcfb38ed031cf6f673/vertical-pod-autoscaler/vendor/k8s.io/klog/v2/k8s_references.go#L93-L98) and adapted the `VpaID`/`PodID` usages to it: see https://github.com/kubernetes/autoscaler/pull/6909/commits/1739a71d80acba2f0ccacd741fd44b747c2b407b